### PR TITLE
Reuse ssh connections during all cmd execution + fix timeouts + fix intermitent connection errors

### PR DESCRIPTION
**Pull request by felipemadero**
Closes https://github.com/ava-labs/avalanche-cli/issues/1219 https://github.com/ava-labs/avalanche-cli/issues/1220 https://github.com/ava-labs/avalanche-cli/issues/1216 https://github.com/ava-labs/avalanche-cli/issues/1217 https://github.com/ava-labs/avalanche-cli/issues/1218

### CODE REVIEW

File: cmd/nodecmd/validateSubnet.go [hunk 1]
```diff
@@ -189,78 +194,68 @@ func validateSubnet(_ *cobra.Command, args []string) error {
 	if blockchainID == ids.Empty {
 		return ErrNoBlockchainID
 	}
-	ansibleNodeIDs, err := ansible.GetAnsibleHostsFromInventory(app.GetAnsibleInventoryDirPath(clusterName))
-	if err != nil {
-		return err
-	}
-	nodeIDMap, failedNodesMap := getNodeIDs(ansibleNodeIDs)
+	nodeIDMap, failedNodesMap := getNodeIDs(hosts)
 	failedNodes := []string{}
```
Comment (arturrez) [inline cmd/nodecmd/validateSubnet.go:198]: replace `failedNodes := []string{} nodeErrors := []error{}` with `map[string]error` no need to have 2 arrays when you can have one map

File: pkg/models/host.go [hunk 2]
```diff
@@ -73,10 +59,16 @@ func (h *Host) GetCloudID() string {
...
-	h.Connection = NewHostConnection(*h, timeout)
-	if !h.Connected() {
-		return fmt.Errorf("failed to connect to host %s", h.IP)
+func (h *Host) Connect() error {
+	if h.Connection != nil {
+		return nil
+	}
+	var err error
+	for i := 0; h.Connection == nil && i < sshConnectionRetries; i++ {
+		h.Connection, err = NewHostConnection(h)
```
Comment (arturrez) [inline pkg/models/host.go:68]: we need to put sleep on exponensial sleep in between

File: pkg/models/host.go [hunk 3]
```diff
@@ -86,43 +78,94 @@ func (h *Host) Connected() bool {
...
-		if err := h.Connect(constants.SSHScriptTimeout); err != nil {
+		if err := h.Connect(); err != nil {
 			return err
 		}
 	}
 	if err := os.MkdirAll(filepath.Dir(localFile), os.ModePerm); err != nil {
 		return err
 	}
-	return h.Connection.Client.Download(remoteFile, localFile)
+	var err error
```
Comment (arturrez) [inline pkg/models/host.go:121]: convert this into func like runWithTimeout where you pass a func and wrap it around context and channel to keep code DRY

**Review by arturrez — APPROVED** [review]

File: pkg/models/host.go (hunk 2 above)
Comment (felipemadero) [reply pkg/models/host.go:68]: No need for sleep. Connection attempt already takes a lot of time and this works pretty fine.

File: pkg/ssh/ssh.go [hunk 4]
```diff
@@ -79,133 +82,230 @@ func PostOverSSH(host models.Host, path string, requestBody string) ([]byte, err
...
+		cloudServerSubnetPath,
+		constants.SSHFileOpsTimeout,
+	)
 }
 
 // RunSSHExportSubnet exports deployed Subnet from local machine to cloud server
 // targets a specific host ansibleHostID in ansible inventory file
-func RunSSHTrackSubnet(host models.Host, subnetName, importPath, networkFlag string) error {
-	return RunOverSSH("Track Subnet", host, "shell/trackSubnet.sh", scriptInputs{SubnetName: subnetName, SubnetExportFileName: importPath, NetworkFlag: networkFlag})
+func RunSSHTrackSubnet(host *models.Host, subnetName, importPath, networkFlag string) error {
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply pkg/ssh/ssh.go:236]: I am fine either way too. That belongs to previous ansible->ssh PRs. Probably it gives a little more readability to the caller code.

File: pkg/ssh/ssh.go [hunk 5]
```diff
@@ -79,133 +82,230 @@ func PostOverSSH(host models.Host, path string, requestBody string) ([]byte, err
...
+		"shell/trackSubnet.sh",
+		scriptInputs{SubnetName: subnetName, SubnetExportFileName: importPath, NetworkFlag: networkFlag},
+	)
 }
 
 // RunSSHUpdateSubnet runs avalanche subnet join <subnetName> in cloud server using update subnet info
-func RunSSHUpdateSubnet(host models.Host, subnetName, importPath string) error {
-	return RunOverSSH("Track Subnet", host, "shell/updateSubnet.sh", scriptInputs{SubnetName: subnetName, SubnetExportFileName: importPath})
+func RunSSHUpdateSubnet(host *models.Host, subnetName, importPath string) error {
+	return RunOverSSH(
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply pkg/ssh/ssh.go:248]: I vote to keep this. At least for this PR matter (but also in general :) )

File: cmd/nodecmd/validateSubnet.go (hunk 1 above)
Comment (felipemadero) [reply cmd/nodecmd/validateSubnet.go:198]: This don't believe for current PR, but will do it

**Pull request merged**